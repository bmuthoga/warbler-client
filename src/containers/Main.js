import React from 'react';
import { 
  Switch, 
  Route, 
  withRouter,
  Redirect 
} from 'react-router-dom'
import { connect } from 'react-redux'
import Homepage from '../components/Homepage'
import AuthForm from '../components/AuthForm'
import { authUser } from '../store/actions/auth'
import { removeError } from '../store/actions/errors'
import MessageForm from './MessageForm'
import withAuth from '../hocs/withAuth'

export const Main = props => {
  const { authUser, errors, removeError, currentUser } = props

  return (
    <div className='container'>
      <Switch>
        <Route 
          exact 
          path='/' 
          render={props => <Homepage { ...props } currentUser={currentUser} />}
        />
        <Route
          exact
          path='/signin'
          render={props => (
            <AuthForm 
              { ... props } 
              buttonText='Log in' 
              heading='Welcome Back' 
              onAuth={authUser}
              errors={errors}
              removeError={removeError}
            />
          )}
        />
        <Route
          exact
          path='/signup'
          render={props => (
            <AuthForm 
              { ... props } 
              buttonText='Sign me up!' 
              heading='Join Warbler Today' 
              onAuth={authUser} 
              errors={errors}
              removeError={removeError}
              signUp
            />
          )}
        />
        <Route
          path='/users/:id/messages/new'
          component={withAuth(MessageForm)}
        />
      </Switch>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    currentUser: state.currentUser,
    errors: state.errors
  }
}

export default withRouter(
  connect(mapStateToProps, { authUser, removeError })(Main)
)
