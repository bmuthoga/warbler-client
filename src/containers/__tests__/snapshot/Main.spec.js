import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Main from '../../Main'

describe('Main Snapshot Tests', () => {
  it('should render the Main component', () => {
    const wrapper = shallow(
      <Main />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
