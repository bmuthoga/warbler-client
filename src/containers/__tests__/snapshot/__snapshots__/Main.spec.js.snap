// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Snapshot Tests should render the Main component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
